pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/BitSlicedRangeIndexReader.java

```diff
@@ -49,49 +48,85 @@ public BitSlicedRangeIndexReader(PinotDataBuffer dataBuffer,
     _min = dataBuffer.getLong(offset);
     offset += Long.BYTES;
     _offset = offset;
-    _max = metadata.hasDictionary()
-        ? metadata.getCardinality() - 1
-        : ((Number) metadata.getMaxValue()).longValue();
+    // TODO: Read max from header to prevent cases where max value is not available in the column metadata
+    if (metadata.hasDictionary()) {
+      _max = metadata.getCardinality() - 1;
+    } else {
+      Number maxValue = (Number) metadata.getMaxValue();
+      _max = maxValue != null ? maxValue.longValue() : Long.MAX_VALUE;
+    }
     _numDocs = metadata.getTotalDocs();
   }
 
   @Override
   public int getNumMatchingDocs(int min, int max) {
+    // TODO: Handle this before reading the range index
+    if (min > max || min > _max || max < _min) {
+      return 0;
+    }
```

Comment:
The check for `min > _max || max < _min` is critical and can only be performed here for raw columns. However, ranges with `min > max` making it down to this level is a problem with the query planner and should be fixed there (even on the broker)

Reply:
Ideally we should also handle `min > _max || max < _min` using the column metadata during the planning

Reply:
Indeed, I don’t think the index should need to do these checks at all, but if you need a quick fix maybe this is the quick solution.